cmd/ecloud/ecloud_vip.go

```diff
@@ -0,0 +1,237 @@
+package ecloud
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/ukfast/cli/internal/pkg/factory"
+	"github.com/ukfast/cli/internal/pkg/helper"
+	"github.com/ukfast/cli/internal/pkg/output"
+	"github.com/ukfast/sdk-go/pkg/service/ecloud"
+)
+
+func ecloudVIPRootCmd(f factory.ClientFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "vip",
+		Short: "sub-commands relating to VIPs",
+	}
+
+	// Child commands
+	cmd.AddCommand(ecloudVIPListCmd(f))
+	cmd.AddCommand(ecloudVIPShowCmd(f))
+	cmd.AddCommand(ecloudVIPCreateCmd(f))
+	cmd.AddCommand(ecloudVIPUpdateCmd(f))
+	cmd.AddCommand(ecloudVIPDeleteCmd(f))
+
+	return cmd
+}
+
+func ecloudVIPListCmd(f factory.ClientFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "Lists VIPs",
+		Long:    "This command lists VIPs",
+		Example: "ukfast ecloud vip list",
+		RunE:    ecloudCobraRunEFunc(f, ecloudVIPList),
+	}
+
+	cmd.Flags().String("name", "", "Name for filtering")
+	cmd.Flags().String("vpc", "", "VPC ID for filtering")
+
+	return cmd
+}
+
+func ecloudVIPList(service ecloud.ECloudService, cmd *cobra.Command, args []string) error {
+	params, err := helper.GetAPIRequestParametersFromFlags(cmd,
+		helper.NewStringFilterFlagOption("name", "name"),
+		helper.NewStringFilterFlagOption("load-balancer", "load_balancer_id"),
+	)
+	if err != nil {
+		return err
+	}
+
+	vips, err := service.GetVIPs(params)
+	if err != nil {
+		return fmt.Errorf("Error retrieving VIPs: %s", err)
+	}
+
+	return output.CommandOutput(cmd, OutputECloudVIPsProvider(vips))
+}
+
+func ecloudVIPShowCmd(f factory.ClientFactory) *cobra.Command {
+	return &cobra.Command{
+		Use:     "show <vip: id>...",
+		Short:   "Shows an VIP",
+		Long:    "This command shows one or more VIPs",
+		Example: "ukfast ecloud vip show vip-abcdef12",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errors.New("Missing VIP")
+			}
+
+			return nil
+		},
+		RunE: ecloudCobraRunEFunc(f, ecloudVIPShow),
+	}
+}
+
+func ecloudVIPShow(service ecloud.ECloudService, cmd *cobra.Command, args []string) error {
+	var vips []ecloud.VIP
+	for _, arg := range args {
+		vip, err := service.GetVIP(arg)
+		if err != nil {
+			output.OutputWithErrorLevelf("Error retrieving VIP [%s]: %s", arg, err)
+			continue
+		}
+
+		vips = append(vips, vip)
+	}
+
+	return output.CommandOutput(cmd, OutputECloudVIPsProvider(vips))
+}
+
+func ecloudVIPCreateCmd(f factory.ClientFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   "Creates a VIP",
+		Long:    "This command creates a VIP",
+		Example: "ukfast ecloud vip create --vpc vpc-abcdef12 --availability-zone az-abcdef12 --spec vips-abcdef12",
```

Comment:
The example here doesn't match available flags

Reply:
Good spot - fixed :)